### PR TITLE
feat: mini-toc changes for mobile exlm-1368

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1368

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live/en/articles/experience-manager/sample-article
- After: https://feat-exlm-1368--exlm--adobe-experience-league.hlx.live/en/articles/experience-manager/sample-article

### CODE REVIEW

**Review by noormd08 — APPROVED** [review]

**Review by noormd08 — CHANGES_REQUESTED** [review]
Can’t we reuse the dropdown script which we created earlier?

**Comment by nitin-rachabathuni** [comment]
> Can’t we reuse the dropdown script which we created earlier?


For Mini TOC mobile/tab we cannot use the existing dropdown script, as look/feel/behaviour is different here Noor.

File: scripts/dropdown/dropdown.js [hunk 1]
```diff
@@ -3,6 +3,11 @@ import { htmlToElement } from '../scripts.js';
 
 loadCSS(`${window.hlx.codeBasePath}/scripts/dropdown/dropdown.css`);
 
+export const DROPDOWN_VARIANTS = {
+  DEFAULT: 'default',
+  MINI_TOC: 'mini-toc',
+};
```
Comment (noormd08) [inline scripts/dropdown/dropdown.js:9]: Can you please the generic name in the variant instead of MINI_TOC

File: scripts/dropdown/dropdown.js [hunk 2]
```diff
@@ -11,12 +16,14 @@ export default class Dropdown {
    * @param {Array} defaultValues - Array of Dropdown default value
    * @param {Array} optionsArrays -  Array of options list
    * @param {Number} id - Unique dropdown id
+   * @param {String} variant - Dropdown variant
    */
-  constructor(parentFormElement, defaultValue, optionsArray) {
+  constructor(parentFormElement, defaultValue, optionsArray, variant) {
```
Comment (noormd08) [inline scripts/dropdown/dropdown.js:21]: Please default variant here itself

File: blocks/mini-toc/mini-toc.css [hunk 3]
```diff
@@ -11,17 +11,18 @@
...
   overflow: visible;
+  max-width: 100%;
 }
 
 .mini-toc-wrapper .mini-toc {
   font-size: var(--spectrum-font-size-50);
   font-weight: var(--spectrum-extra-bold-font-weight);
   padding-left: 0;
-  overflow: hidden;
+  overflow: visible;
```
Comment (noormd08) [inline blocks/mini-toc/mini-toc.css:21]: Any reason for making visible? Ideally it should be auto in case we want to have a scroll on need basis right?

Comment (nitin-rachabathuni) [reply blocks/mini-toc/mini-toc.css:21]: This is for mobile where we have to see the internal contents of the mini-toc in form of a dropdown. We are overriding it for higher resolutions.
And this is not working with `overflow: auto`

File: scripts/dropdown/dropdown.js (hunk 1 above)
Comment (nitin-rachabathuni) [reply scripts/dropdown/dropdown.js:9]: done Noor

File: scripts/dropdown/dropdown.js (hunk 2 above)
Comment (nitin-rachabathuni) [reply scripts/dropdown/dropdown.js]: done done

**Review by noormd08 — APPROVED** [review]

**Pull request merged**